src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -266,27 +285,50 @@ private AddInterfaceInteraction createAddInterfaceInteractionAction(@Nonnull Pay
                                 // 3. add paypal response to the ctp payment as interaction interface
-                                return ctpFacade.getPaymentService().updatePayment(payment, Collections.singletonList(addInteractionAction))
+                                return updateCtpPayment(paypalPayment, updatedCtpPayment)
                                         // 4. create charge transaction in the ctp payment
-                                        .thenApply(updatedCtpPayment -> createChargeTransaction(paypalPlusPaymentId, paypalPayment, updatedCtpPayment));
+                                        .thenApply(updatedCtpPayment2 -> createChargeTransaction(paypalPlusPaymentId, paypalPayment, updatedCtpPayment2));
                             });
                 })
                 .thenApply(ignore -> PaymentHandleResponse.ofHttpStatus(HttpStatus.CREATED));
     }
 
+    private CompletionStage<io.sphere.sdk.payments.Payment> updateCtpPayment(Payment paypalPayment, io.sphere.sdk.payments.Payment ctpPayment) {
+        ImmutableList.Builder<UpdateAction<io.sphere.sdk.payments.Payment>> builder = ImmutableList.builder();
+        builder.add(createAddInterfaceInteractionAction(paypalPayment, RESPONSE));
+        PaymentInstruction paymentInstruction = paypalPayment.getPaymentInstruction();
+        if (paymentInstruction != null) {
+            builder.addAll(createAddPaymentInstructionAction(paymentInstruction));
+        }
+        return ctpFacade.getPaymentService().updatePayment(ctpPayment, builder.build());
+    }
+
+    private List<SetCustomField> createAddPaymentInstructionAction(PaymentInstruction paymentInstruction) {
+        ImmutableList.Builder<SetCustomField> listBuilder = ImmutableList.builder();
+        listBuilder.add(SetCustomField.ofObject(REFERENCE, paymentInstruction.getReferenceNumber()));
+        listBuilder.add(SetCustomField.ofObject(BANK_NAME, paymentInstruction.getRecipientBankingInstruction().getBankName()));
+        listBuilder.add(SetCustomField.ofObject(ACCOUNT_HOLDER_NAME, paymentInstruction.getRecipientBankingInstruction().getAccountHolderName()));
+        listBuilder.add(SetCustomField.ofObject(IBAN, paymentInstruction.getRecipientBankingInstruction().getInternationalBankAccountNumber()));
+        listBuilder.add(SetCustomField.ofObject(BIC, paymentInstruction.getRecipientBankingInstruction().getBankIdentifierCode()));
+        listBuilder.add(SetCustomField.ofObject(PAYMENT_DUE_DATE, paymentInstruction.getPaymentDueDate()));
+        listBuilder.add(SetCustomField.ofObject(AMOUNT, Money.of(new BigDecimal(paymentInstruction.getAmount().getValue()),
+                paymentInstruction.getAmount().getCurrency())));
+        return listBuilder.build();
+    }
+
     private CompletionStage<io.sphere.sdk.payments.Payment> createChargeTransaction(@Nonnull String paypalPlusPaymentId,
```

Comment:
pls, put the overloaded methods (`createChargeTransaction`) next to each other in the class file.

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/51/commits/e838e31a00f58885521f23850ff2cfbf07a26d8c

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/51/commits/6cb6c76fa37871969a1aae7f2c2937e13c7265be